pkgs/by-name/ho/homebox/package.nix

```diff
@@ -0,0 +1,145 @@
+{
+  stdenvNoCC,
+  jq,
+  moreutils,
+  nodePackages,
+  cacert,
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}: let
+  pname = "homebox";
+  version = "0.10.3";
+  src = fetchFromGitHub {
+    owner = "hay-kot";
+    repo = "homebox";
+    rev = "v${version}";
+    hash = "sha256-Hej/dM0BgtRWiMOpp/SDVr3H1IbYb935T1pfX8apjpE=";
+    # Inspired by: https://github.com/NixOS/nixpkgs/blob/f2d7a289c5a5ece8521dd082b81ac7e4a57c2c5c/pkgs/applications/graphics/pdfcpu/default.nix#L20-L32
+    # The intention here is to write the information into files in the `src`'s
+    # `$out`, and use them later in other phases (in this case `preBuild`).
+    # In order to keep determinism, we also delete the `.git` directory
+    # afterwards, imitating the default behavior of `leaveDotGit = false`.
+    # More info about git log format can be found at `git-log(1)` manpage.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git log -1 --pretty=%H > "backend/COMMIT"
+      git log -1 --pretty=%cd --date=format:'%Y-%m-%dT%H:%M:%SZ' > "backend/SOURCE_DATE"
+      rm -rf ".git"
+    '';
+  };
+  unsupported = throw "Unsupported system ${stdenvNoCC.hostPlatform.system} for ${pname}";
+
+  frontend = stdenvNoCC.mkDerivation {
+    pname = "${pname}-frontend";
+    inherit version;
+
+    src = "${src}/frontend";
+
+    preBuild = ''
+      export HOME=$(mktemp -d)
+      export STORE_PATH=$(mktemp -d)
+
+      pnpm config set store-dir "${pnpm-deps}"
+      pnpm install --offline --frozen-lockfile --shamefully-hoist
+      patchShebangs node_modules/{*,.*}
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+
+      pnpm build
+
+      runHook postBuild
+    '';
+    env.NUXT_TELEMETRY_DISABLED = 1;
+
+    nativeBuildInputs = [
+      nodePackages.pnpm
+    ];
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      cp -r .output/public/* $out/
+
+      runHook postInstall
+    '';
+  };
+  pnpm-deps = stdenvNoCC.mkDerivation {
+    pname = "${pname}-pnpm-deps";
+    inherit version;
+    src = "${src}/frontend";
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      nodePackages.pnpm
+      cacert
+    ];
+
+    installPhase = ''
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      # This version of the package has different versions of esbuild as a dependency.
+      # You can use the command below to get esbuild binaries for a specific platform and calculate hashes for that platforms. (linux, darwin for os, and x86, arm64, ia32 for cpu)
+      # cat package.json | jq '.pnpm.supportedArchitectures += { "os": ["linux"], "cpu": ["arm64"] }' | sponge package.json
+      pnpm install --frozen-lockfile --ignore-script
```

Comment:
Please use the pnpm helper from nixpkgs which prunes such binaries and creates one hash for all binaries. We can then use our esbuild and overwrite the version if necessary

Reply:
While I was able to use the helper, and it made some things easier, the package needs `pnpm install` to be called with the `--shamefully-hoist` option, which the helper does not. My solution is to call install again with the needed option, which does work, but now we install the pnpm deps twice. Is there any solution to avoid this or is it fine?